js/ccf-app/src/global.ts

```diff
@@ -312,6 +347,40 @@ export interface CCF {
    * Only defined for endpoints with "mode" set to "historical".
    */
   historicalState?: HistoricalState;
+
+  historical: {
+    /**
+     * Retrieve a range of historical states containing the state written at the given
+     * indices.
+     *
+     * If this is not currently available, this function returns `null`
+     * and begins fetching the ledger entry asynchronously. This will generally
+     * be true for the first call for a given seqno, and it may take some time
+     * to completely fetch and validate. The call should be repeated later with
+     * the same arguments to retrieve the requested entries. This state is kept
+     * until it is deleted for one of the following reasons:
+     *  - A call to {@link CCF.historical.dropCachedStateRange}
+     *  - `seconds_until_expiry` seconds elapse without calling this function
+     *  - This handle is used to request a different seqno or range
+     *
+     * The range is inclusive of both start_seqno and end_seqno. If a non-empty
+     * array is returned, it will always contain the full requested range; the
+     * array will be of length (end_seqno - start_seqno + 1).
+     */
+    getStateRange(
+      handle: number,
+      start_seqno: number,
+      end_seqno: number,
+      seconds_until_expiry: number
+    ): HistoricalState[] | null;
+
+    /** Drop cached states for the given handle.
+     *
+     * May be used to free up space once a historical query has been resolved,
+     * more aggressively than waiting for the requests to expire.
+     */
+    dropCachedStateRange(handle: number): void;
```

Comment:
The C++ name is `drop_request`, however I find that very confusing and it also doesn't play well with re-exporting functions in the root module of the ccf-app npm package.